### PR TITLE
feat: [PL-28041]: Add error handler ldap auth UI 

### PR DESCRIPTION
**Pull request by pbarapatre10**
#### Screenshots
Before :
image

After: 

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11149)

**Comment by pbarapatre10** [comment]
retrigger typecheck

**Comment by pbarapatre10** [comment]
retrigger titlecheck

File: src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx [hunk 1]
```diff
@@ -90,34 +89,50 @@ const useLdapTestConfigurationProvider = ({
...
-        showError(ldapTestResource?.message, 5000)
-        return
-      }
-      const { resource: updateAuthResource, responseMessages } = await updateAuthMechanism(undefined, {
-        queryParams: {
-          accountIdentifier: accountId,
-          authenticationMechanism: AuthenticationMechanisms.LDAP
-        }
-      })
+    /* istanbul ignore else */
```
Comment (avinashmadhwani02) [inline src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx:102]: why are we ignoring this from coverage?

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx [hunk 2]
```diff
@@ -144,6 +159,11 @@ const useLdapTestConfigurationProvider = ({
                   inputGroup={{ type: 'password' }}
                   disabled={isDisabled}
                 />
+                <Layout.Horizontal /* className={css.alignCenter} */>
```
Comment (meenakshi-harness) [inline src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx:162]: remove commented code

File: src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx [hunk 3]
```diff
@@ -90,34 +89,50 @@ const useLdapTestConfigurationProvider = ({
...
+        /* istanbul ignore next */ showError(e.data?.message || e.message, 5000)
       }
-    } catch (e) {
-      /* istanbul ignore next */ showError(e.data?.message || e.message, 5000)
     }
   }
+
+  const resetStatesAndInvokeLdapAuth = async (formData: FormData): Promise<boolean> => {
+    setErrorMessages(null)
+    const response = await ldapLoginTest(formData as any)
```
Comment (meenakshi-harness) [inline src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx:127]: why any?

File: src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx (hunk 1 above)
Comment (pbarapatre10) [reply src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx:102]: We already have a test for the case of if (test('LDAP Provider enable config')), but on else we won't be able to enable LDAP auth mechanism, so no way to test it as we have no action there. Thanks

File: src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx (hunk 2 above)
Comment (pbarapatre10) [reply src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx]: done!

File: src/modules/33-auth-settings/pages/Configuration/LDAPProvider/__tests__/LDAPProvider.test.tsx [hunk 4]
```diff
@@ -307,7 +333,7 @@ describe('LDAP Provider', () => {
     await act(async () => {
       testLdapConfigBtn && fireEvent.click(testLdapConfigBtn)
     })
-    await waitFor(() => expect(getByText('authSettings.ldap.ldapTestSuccessful')).not.toBeNull())
+    expect(getAllByText('authSettings.ldap.ldapTestSuccessful')).toBeDefined()
```
[the remark this answers is not in the document]
Comment (pbarapatre10) [reply src/modules/33-auth-settings/pages/Configuration/LDAPProvider/__tests__/LDAPProvider.test.tsx]: added back.

File: src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx [hunk 5]
```diff
@@ -69,17 +71,14 @@ const useLdapTestConfigurationProvider = ({
...
-      if (resource?.status != 'SUCCESS') {
-        showError(resource?.message, 5000)
-      } else {
-        showSuccess(getString('authSettings.ldap.ldapTestSuccessful'), 5000)
-      }
-    } catch (e) {
-      /* istanbul ignore next */ showError(e.data?.message || e.message, 5000)
+      await resetStatesAndInvokeLdapAuth(formData)
+    } catch (e: any) /* istanbul ignore next */ {
+      setErrorMessages(defaultTo(e.data?.responseMessages, [{ level: 'ERROR', message: e.data?.message || e.message }]))
```
[the remark this answers is not in the document]
Comment (pbarapatre10) [reply src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx:76]: We are using <ErrorHandler> component, & I think the `getErrorInfoFromErrorObject` is mostly used with Toaster. 
If possible to use this, we will take it up in next CL. Thanks

File: src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx (hunk 3 above)
Comment (pbarapatre10) [reply src/modules/33-auth-settings/modals/LdapTestConfiguration/useLdapTestConfigurationProvider.tsx:127]: The `usePostLdapAuthenticationTest` takes @FormDataParam in RequestBody and not a RequestType object, so have to use 'any' here

**Review by meenakshi-harness — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Pull request merged**